lib/katello/tasks/upgrades/3.11/update_puppet_repos.rake

```diff
@@ -5,10 +5,16 @@ namespace :katello do
       task :update_puppet_repos => %w(environment) do
         User.current = User.anonymous_admin
         Katello::Repository.puppet_type.each do |repo|
-          puts "Refreshing repository #{repo.label} (#{repo.id})"
-          install_dist = repo.backend_service(SmartProxy.pulp_master).backend_data['distributors'].find { |dist| dist['distributor_type_id'] == 'puppet_install_distributor' }
-          SmartProxy.pulp_master.pulp_api.resources.repository.delete_distributor(repo.pulp_id, install_dist['id'])
-          ForemanTasks.sync_task(::Actions::Pulp::Repository::Refresh, repo)
+          begin
+            puts "Refreshing repository #{repo.label} (#{repo.id})"
+            install_dist = repo.backend_service(SmartProxy.pulp_master).backend_data['distributors'].find { |dist| dist['distributor_type_id'] == 'puppet_install_distributor' }
+            if install_dist
+              SmartProxy.pulp_master.pulp_api.resources.repository.delete_distributor(repo.pulp_id, install_dist['id'])
+            end
+            ForemanTasks.sync_task(::Actions::Pulp::Repository::Refresh, repo)
+          rescue RestClient::ResourceNotFound
```

Comment:
does the 404 ever happen as a result of normal-ish data?

Reply:
it should not no, and if a repo is missing in pulp, it should get corrected by the correct_repos step/task anyways, but i figured it'd be better to handle it here as well

Reply:
I would log a warning here that pulp_id was not found before continuing